From 552222973b1ac23cf5fd03a6ef0d1610baa7a911 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Mon, 18 Sep 2017 14:08:57 +0800
Subject: [PATCH 2556/5242] MLK-16285-6 usb: cdns3: add more operatons at
 cdns3_remove

commit  9e71ef5fb297d27a7bea9ff2a6c99e2168dc4d25 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add PHY shutdown and clock disable operations at cdns3_remove.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/core.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/cdns3/core.c b/drivers/usb/cdns3/core.c
index f977465..b640197 100644
--- a/drivers/usb/cdns3/core.c
+++ b/drivers/usb/cdns3/core.c
@@ -562,6 +562,8 @@ static int cdns3_remove(struct platform_device *pdev)
 	struct cdns3 *cdns = platform_get_drvdata(pdev);
 
 	cdns3_remove_roles(cdns);
+	usb_phy_shutdown(cdns->usbphy);
+	cdns3_disable_unprepare_clks(&pdev->dev);
 
 	return 0;
 }
-- 
1.7.9.5

